keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -215,6 +225,113 @@ class SqlArtifactRepository(
+//          where da.uid = eav.artifact_uid
+//          and da.name = 'keel'
+//          and da.type = 'DEB'
+//          and e.uid = eav.environment_uid
+//          and e.delivery_config_uid = dc.uid
+//          and dc.name = 'keel-manifest'
+//          union
+//          select e.name, dav.version, null, null
+//          from delivery_artifact_version dav, delivery_artifact da, environment e, delivery_config dc
+//          where da.uid = dav.delivery_artifact_uid
+//          and da.name = 'keel'
+//          and da.type = 'DEB'
+//          and e.delivery_config_uid = dc.uid
+//          and dc.name = 'keel-manifest'
+//          and not exists (
+//          select 1 from environment_artifact_versions eav
+//          where eav.artifact_version = dav.version
+//          and eav.environment_uid = e.uid
+//        )
+        val rawResults = jooq
+          .select(
+            ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION,
+            ENVIRONMENT_ARTIFACT_VERSIONS.APPROVED_AT,
+            ENVIRONMENT_ARTIFACT_VERSIONS.DEPLOYED_AT
+          )
+          .from(
+            ENVIRONMENT_ARTIFACT_VERSIONS,
+            DELIVERY_ARTIFACT,
+            ENVIRONMENT,
+            DELIVERY_CONFIG
+          )
+          .where(DELIVERY_ARTIFACT.UID.eq(ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_UID))
+          .and(DELIVERY_ARTIFACT.NAME.eq(artifact.name))
+          .and(DELIVERY_ARTIFACT.TYPE.eq(artifact.type.name))
+          .and(ENVIRONMENT.UID.eq(ENVIRONMENT_ARTIFACT_VERSIONS.ENVIRONMENT_UID))
+          .and(ENVIRONMENT.DELIVERY_CONFIG_UID.eq(DELIVERY_CONFIG.UID))
+          .and(DELIVERY_CONFIG.NAME.eq(deliveryConfig.name))
+          .and(ENVIRONMENT.NAME.eq(environment.name))
+          .unionAll(
+            select(
+              DELIVERY_ARTIFACT_VERSION.VERSION,
+              castNull(LocalDateTime::class.java),
+              castNull(LocalDateTime::class.java)
+            )
+              .from(
+                DELIVERY_ARTIFACT_VERSION,
+                DELIVERY_ARTIFACT,
+                ENVIRONMENT,
+                DELIVERY_CONFIG
+              )
+              .where(DELIVERY_ARTIFACT.UID.eq(DELIVERY_ARTIFACT_VERSION.DELIVERY_ARTIFACT_UID))
+              .and(DELIVERY_ARTIFACT.NAME.eq(artifact.name))
+              .and(DELIVERY_ARTIFACT.TYPE.eq(artifact.type.name))
+              .and(ENVIRONMENT.DELIVERY_CONFIG_UID.eq(DELIVERY_CONFIG.UID))
+              .and(DELIVERY_CONFIG.NAME.eq(deliveryConfig.name))
+              .and(ENVIRONMENT.NAME.eq(environment.name))
+              .andNotExists(
+                selectOne()
+                  .from(ENVIRONMENT_ARTIFACT_VERSIONS)
+                  .where(ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION.eq(DELIVERY_ARTIFACT_VERSION.VERSION))
+                  .and(ENVIRONMENT_ARTIFACT_VERSIONS.ENVIRONMENT_UID.eq(ENVIRONMENT.UID))
+              )
+          )
+          .fetch()
+        val turnt = rawResults.map { (version, approvedAt, deployedAt) ->
```

Comment:
what does `turnt` mean?

Reply:
It means I forgot to inline some variables